oci/client/build.go

```diff
@@ -32,12 +31,34 @@ import (
 
 // Build archives the given directory as a tarball to the given local path.
 // While archiving, any environment specific data (for example, the user and group name) is stripped from file headers.
-func (c *Client) Build(artifactPath, sourceDir string, ignorePaths []string) (err error) {
-	if f, err := os.Stat(sourceDir); os.IsNotExist(err) || !f.IsDir() {
-		return fmt.Errorf("invalid source dir path: %s", sourceDir)
+func (c *Client) Build(artifactPath, source string, ignorePaths []string) (err error) {
+	if f, err := os.Stat(source); os.IsNotExist(err) || !f.IsDir() {
+		// if the *source* is a file, we create a temp directory and write *source* file to it,
+		// then we replace *source* parameter with the temp directory path to treat *source* as if it were a folder
+		tmp, err := os.MkdirTemp("", "oci_copy")
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(tmp)
+
+		f, err := os.ReadFile(source)
+		if err != nil {
+			return err
+		}
+
+		err = os.WriteFile(filepath.Join(tmp, filepath.Base(source)), f, 0644)
+		if err != nil {
+			return err
+		}
+
+		if err := os.Rename(filepath.Join(tmp, filepath.Base(source)), filepath.Join("./", source)); err != nil {
```

Comment:
Are we copying the file to temp and then we move it from tmp to current directory? Instead of doing this let's just feed the file stream to the tar one please.